client/src/components/overview/overview.jsx

```diff
@@ -38,13 +126,71 @@ class Overview extends React.Component {
         <div className="Slogan">{this.state.info.slogan}</div>
         <div className="Description">{this.state.info.description}</div>
         <br></br>
-        {renderFeatures}
+        <NewStyles
+          showWarning={this.state.showWarning}
+          allStyles={this.state.styles.results}
+          selectedStyle={this.state.selectedStyle}
+          setSelectedStyle={this.setSelectedStyle}
+          currentStyleInfo={this.state.currentStyleInfo}
+          setCurrentStyleInfo={this.setCurrentStyleInfo}
+          setChosenSize={this.setChosenSize}
+          setMainPhoto={this.setMainPhoto}
+          setThumbnails={this.setThumbnails}
+          setAvailableSizes={this.setAvailableSizes}
+          setAvailableAmount={this.setAvailableAmount}
+          setSkuID={this.setSkuID}
+          setChosenAmount={this.setChosenAmount}
+
+          original_price={this.state.original_price}
+          sale_price={this.state.sale_price}
+          setOriginalPrice={this.setOriginalPrice}
+          setSalePrice={this.setSalePrice}
+        />
         <br></br>
-        {renderStyles}
+        <Modal
+          showModal={this.state.showModal}
+          setShowModal={this.setShowModal}
+          currentStyleInfo={this.state.currentStyleInfo}
+        />
         <br></br>
+        <NewPhotos
+          currentStyleInfo={this.state.currentStyleInfo}
+          mainPhoto={this.state.mainPhoto}
+          setMainPhoto={this.setMainPhoto}
+          thumbnails={this.state.thumbnails}
+          setThumbnails={this.setThumbnails}
+          showModal={this.state.showModal}
+          setShowModal={this.setShowModal}
+          thumbnailIndex={this.state.thumbnailIndex}
+          setThumbnailIndex={this.setThumbnailIndex}
+        />
+        <br></br>
+        <NewSize
+          chosenSize={this.state.chosenSize}
+          chosenAmount={this.state.chosenAmount}
+          availableSizes={this.state.availableSizes}
+          setChosenSize={this.setChosenSize}
+          setAvailableAmount={this.setAvailableAmount}
+          setSkuID={this.setSkuID}
+          setChosenAmount={this.setChosenAmount}
+        />
+        <br></br>
+        <NewAmount
+          chosenSize={this.state.chosenSize}
+          availableAmount={this.state.availableAmount}
+          setChosenAmount={this.setChosenAmount}
+          chosenAmount={this.state.chosenAmount}
+        />
+        <br></br>
+        <NewButton
+          chosenSize={this.state.chosenSize}
+          chosenAmount={this.state.chosenAmount}
+          skuID={this.state.skuID}
+          showWarning={this.state.showWarning}
+          availableAmount={this.state.availableAmount}
+        />
       </div>
     );
   }
 }
 export default Overview;
```

Comment:
This Overview file is huge, but more than that it is brilliant. I really love the way it is organized. Makes me want to redo all of my stuff... lol.

Reply:
I'm glad you think that way! I don't think it's 100% recommended during job hunts, but this philosophy really has some good trade offs to offer.